src/components/widget_types/twitter.vue

```diff
@@ -54,15 +57,14 @@
             sentData: function() {
 
             },
-            "api.NumberTweets": function() {
-                this.numberTweets = this.api.NumberTweets
-            },
-            "api.APIKey": function() {
-                this.apiKey = this.api.APIKey
-            },
-            "api.UserId": function() {
-                this.userId = this.api.userId
-            },
+            api : function() {
+              this.numberTweets = this.api.NumberTweets;
+              this.consumerKey = this.api.ConsumerKey;
+              this.consumerSecret = this.api.ConsumerSecret;
+              this.accessToken = this.api.AccessToken;
+              this.accessSecret = this.api.AccessSecret;
+              this.userId = this.api.UserId;
+            }
```

Comment:
Apparently, we can just put it all in a watch for api and it gets called if any of them change. Sorry for doubting you @ConnorBach

Reply:
Oh nice, no worries!